### ISSUE TITLE
requestBody is not really optional (unsupported media type undefined)

### ISSUE THREAD

**Issue by ownagedj** [cdimascio/express-openapi-validator#397]
**Describe the bug**
If I define an operation with an optional requestBody and I do not send any `content-type` header nor content then the request will result in an 415 error.

**To Reproduce**
See PR

**Actual behavior** 
The requestBody is required even though it is optional

**Expected behavior**
The request should succeed without a requestBody

**Examples and context**
See PR

PR: https://github.com/cdimascio/express-openapi-validator/pull/398

**Comment by cdimascio** [comment]
@ownagedj thanks for your fix. its available in `v4.2.1`

**Comment by Saniyook** [comment]
Hi @cdimascio, this bug is still reproduced in the current version. (4.7.1). :(

**Comment by cdimascio** [comment]
Can you post a canonical snippet of the spec that is causing the error. I'll have a look

**Comment by jowilhnson** [comment]
I just ran into this same issue using spec 3.1.0 and both express-openapi-validator 5.4.2 and 5.5.7

```
post:
  requestBody:
    content:
      application/json:
        schema:
          type: object
```

Works fine when I pass a request body, but when I don't I get the 415 error with "unsupported media type undefined".

I've poured through the openapi docs but haven't found any other way to specify that the request body can either be the specified json object, or nothing at all. And the spec states requestBody is optional by default, so it seems like as long as I don't use required, it shouldn't give this error. 

The only workaround I've found is to always send an object in the request body, and tell the spec to accept either an object with the specified properties, or an object with no properties and additionalProperties: false. Essentially either take an empty object or the object with data. 

This isn't a perfect solution, and having the requestBody actually be optional would be better, but it works well enough for how infrequently I need it.
